compiler/rustc_typeck/src/check/fn_ctxt_checks.rs

```diff
@@ -0,0 +1,975 @@
+use super::coercion::CoerceMany;
```

Comment:
would it make more sense to instead move this into `fn_ctxt/checks.rs`? This seems a bit cleaner to me

Reply:
I like this. More submodules instead of sibling modules with prefixes.

Reply:
Thank you for the idea. I will get that in as soon as I can.